test updates

detiber opened the pull request:
- fix nose coverage flags
- add coverage support for files tested outside of the utils directory

rhcarvalho commented on utils/setup.cfg:
```diff
@@ -7,10 +7,13 @@ universal=1
-cover_min_percentage=70
+with-coverage=1
+cover-html=1
+#cover-package=ooinstall,modify_yaml,openshift_master_facts_default_predicates,openshift_master_facts_default_priorities
```
Is this commented aiming to be uncommented in the future?

detiber replied:
Just something I accidentally left in after testing. Will remove shortly, wanted to wait for testing to complete before I removed it (I noticed after I tagged for testing).

rhcarvalho commented:
@detiber FYI we added Travis to run `make ci` (I think you were on PTO back then), so for this type of changes it might be quickest way to get feedback, without the `a o s - c i - t e s t`.

https://travis-ci.org/openshift/openshift-ansible/builds/181696689

rhcarvalho commented:
Very weird that we got green from the default job, and red from travis:

https://ci.openshift.redhat.com/jenkins/job/openshift-ansible/5234/console
https://travis-ci.org/openshift/openshift-ansible/jobs/181696690

detiber commented:
@rhcarvalho travis pulled in an additional dependency during testing which brought the coverage down, I updated the .coveragerc to exclude it and also removed the commented line.

detiber commented:
aos-ci-test

rhcarvalho commented on utils/.coveragerc:
```diff
@@ -0,0 +1,6 @@
+[run]
+omit=
+    */lib/python*/site-packages/*
```
Is the coverage plugin running against the stdlib by default?!

detiber replied:
It is not.

detiber replied:
@rhcarvalho The main reason is, because removing the package filter causes additional code to be pulled in for the coverage report, including packages installed in the virtualenv, and in some cases stdlib packages. The omit config removes these items from the coverage report.

rhcarvalho commented on utils/.coveragerc:
```diff
@@ -0,0 +1,6 @@
+    */lib/python*/site-packages/*
+    */encodings/ascii.py
+    */getpass.py
+    */contextlib.py
```
What are these 3 meant to exclude?

detiber replied:
They are pulled in as dependencies, the first two lines exclude items that were explicitly installed within the virtualenv...  I'm not sure where getpass.py and contextlib.py are coming in from, but I'll see if I can figure it out.

detiber replied:
I suspect we are hitting an odd bug from nose/coverage.py that is pulling in contextlib and getpass

detiber replied:
I think I found a better workaround, updating the PR.

detiber commented:
aos-ci-test

detiber commented:
@tbielawa ptal

tbielawa commented on utils/setup.cfg:
```diff
@@ -7,10 +7,12 @@ universal=1
+cover-html=1
+cover-inclusive=1
+cover-min-percentage=70
+cover-erase=1
```
WRT: `cover-erase=1` personally I prefer to run coverage with the previous results retained. This allows you to easily see if coverage has increased or decreased in any given PR/commit. I understand why you'd set the value to `1` If that doesn't work on travis (I imagine travis erases all test environments between runs).

detiber replied:
I'm good either way, I was iterating through some issues I was hitting, and cover-erase snuck in as I was trying to make sure that I wasn't reporting data from a previous run.

tbielawa commented:
shippit

On Tue, Dec 6, 2016 at 1:08 PM, Jason DeTiberus <notifications@github.com>
wrote:

> *@detiber* commented on this pull request.
> ------------------------------
>
> In utils/setup.cfg
> <https://github.com/openshift/openshift-ansible/pull/2927>:
>
> > @@ -7,10 +7,12 @@ universal=1
>  [nosetests]
>  tests=../,../roles/openshift_master_facts/test/,test/
>  verbosity=2
> -with_coverage=1
> -cover_html=1
> -cover_package=ooinstall
> -cover_min_percentage=70
> +with-coverage=1
> +cover-html=1
> +cover-inclusive=1
> +cover-min-percentage=70
> +cover-erase=1
>
> I'm good either way, I was iterating through some issues I was hitting,
> and cover-erase snuck in as I was trying to make sure that I wasn't
> reporting data from a previous run.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/openshift-ansible/pull/2927>, or mute the
> thread
> <https://github.com/notifications/unsubscribe-auth/AACBYfNLc42iPQn8_eQynKc41a_w9zz5ks5rFc7VgaJpZM4LFk90>
> .
>



-- 
Tim Bielawa, Software Engineer [ED-C137]
Cell: 919.332.6411  | IRC: tbielawa (#openshift)
1BA0 4FAB 4C13 FBA0 A036  4958 AD05 E75E 0333 AE37

detiber commented:
@rhcarvalho @tbielawa closing this PR, I've included the changes in https://github.com/openshift/openshift-ansible/pull/2892
